components/Editor/InstructionsTable.tsx

```diff
@@ -24,6 +31,84 @@ export const InstructionsTable = ({
   const tableRef = useRef<HTMLDivElement>(null)
   const focusRowRef = useRef<HTMLTableRowElement>(null)
 
+  const splitInLines = (instructionName: string) =>
+    instructionName.split('\n').map((line, i) => (
+      <span key={i}>
+        {i !== 0 && <br />}
+        {line}
+      </span>
+    ))
+
+  const isVariable = (s: string) => variableRe.test(s)
+  const getVariableNameFromTag = (tag: string) => {
+    const match = variableRe.exec(tag)
+    if (!match) {
+      return null
+    }
+
+    return match[1]
+  }
+  const getRandomToolTipId = () =>
+    `tooltip-${Math.floor(Math.random() * 100000000)}`
+
+  const formatVariableValues = (values: Array<string>): string => {
+    // TODO if type info is provided by the back-end
+    // => convert the array of felt252 to a more human-readable
+    // value, according to the variable type.
+    return `[${values.join(', ')}]`
+  }
+
+  const buildVariableSpanWithTooltip = (
+    sierraVariableTag: string,
+    key: number,
+  ) => {
+    const variableName = getVariableNameFromTag(sierraVariableTag)
+    if (!variableName) {
+      return sierraVariableTag
+    }
+
+    if (!(variableName in variables) || variables[variableName].length === 0) {
+      return sierraVariableTag
+    }
+
+    const variableValues = variables[variableName]
+    const tooltipId = getRandomToolTipId()
+
+    return (
+      <>
+        <span
+          key={key}
+          data-tip
+          data-for={tooltipId}
+          className="hover:text-orange-500 cursor-pointer"
+        >
+          {sierraVariableTag}
+        </span>
+        <ReactTooltip id={tooltipId} effect="solid">
+          <span>{formatVariableValues(variableValues)}</span>
+        </ReactTooltip>
+      </>
+    )
+  }
+
+  const formatInstructionWithVariables = (instructionName: string) => {
```

Comment:
Given:
* The InstructionsTable.tsx component is used to render both CASM and Sierra
* AND this function is purpose built for Sierra

I suggest to reflect that in the name, i.e. `formatSierraInstructionWithVariables` or `addVariableTolltipsToSierraInstruction`.

Similarly for other functions and variable names in this file. If they are Sierra specific it would be good to reflect that in the name.

Reply:
Yes, I agree to give more explicit names. I've also added the `codeType` to the `InstructionTable` component to be sure to not polluting CASM instructions rendering with Sierra stuff.